Mv3 security page update (#6584)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/mv3/security/index.md

The Chrome browser limits an extension's access to privileges that have been explicitly requested in
the [manifest][5]. Extensions should minimize their permissions by only registering APIs and
websites they depend on.

Limiting an extension's privileges limits what a potential attacker can exploit.

### Cross-origin fetch() {: #xhr }

> An extension can only use `fetch()` and [XMLHttpRequest][6] to get resources from the
> extension and from domains specified in the permissions. Note that calls to both are intercepted by
> the [fetch][27] handler in the service worker.

Reviewer: ```suggestion
An extension can only use `fetch()` and [`XMLHttpRequest()`][6] to get resources from the
extension and from domains specified in the permissions. Note that calls to both are intercepted by
the [`fetch'][27] handler in the service worker.
```

The Chrome browser limits an extension's access to privileges that have been explicitly requested in
the [manifest][5]. Extensions should minimize their permissions by only registering APIs and
websites they depend on.

Limiting an extensions privileges limits what a potential attacker can exploit.

### Cross-origin fetch() {: #xhr }

An extension service worker can only use `fetch()`, or APIs that use the same [fetch handler][27] to get resources from itself and from domains

> specified in the permissions.

Reviewer: The original wording was correct. This wording is problematic. The only other API that goes through the service worker's `fetch()` handler is `XMLHTTPRequest()`.

### Cross-origin fetch() {: #xhr }

An extension can only use `fetch()` or [XMLHttpRequest][6] to get resources from itself and from domains

> specified in the permissions, as both API's use the same [fetch handler][27] in the service worker.

Reviewer: ```suggestion
When used in an extension `fetch()` and [`XMLHttpRequest()`][6] can only get resources from the
extension and from domains specified in the permissions. Note that calls to both are intercepted by
the [fetch][27] handler in the service worker.
```

  "name": "Very Secure Extension",
  "version": "1.0",
  "description": "Example of a Secure Extension",
   "content_security_policy": {
    "extension_pages": "default-src 'self'"
  },
  "manifest_version": 3
}
```

> If the extension needs to make use of web assembly, or increase the restrictions on [sandboxed pages](/docs/extensions/mv3/sandboxingEval/), they can be added:

Reviewer: ```suggestion
If the extension needs to use web assembly, or increase the restrictions on [sandboxed pages](/docs/extensions/mv3/sandboxingEval/), they can be added:
```

  (e.g., [Spectre][14]), and to being taken over by an attacker if a malicious web page compromises
  the renderer process.

Sensitive work should be performed in a dedicated process, such as the extension's service worker. 

> Avoid accidentally exposing extension privileges to content scripts:

Reviewer: I think this needs more information. It may be more complicated than replacing 'background script' with 'service worker' since the former can do more things than the latter. I don't have a specific suggestion, but I'd start by finding out what 'sensitive work' means.
